### ISSUE TITLE
[Regression] workspace isolation broken with gnome-shell 43.6

### ISSUE THREAD

**Issue by khimaros** [blipk/Customised-Workspaces#47]
### Desktop:
 - Gnome-Shell Version (gnome-shell --version): `43.6`
 - GJS Version (gjs --version): `1.76.2`
 - Extension Version [e.g. 22]: `gnome-extensions info worksets@blipk.xyz | grep Version:
  Version: 59`
 - Linux distribution and version: Debian trixie

### Describe the bug
isolation is not working. after opening a single gnome-terminal app (or files, or firefox, etc. mostly wayland apps?), the app shows on all workspaces and clicking on dock switches to the existing window instead of launching another one.

this has also been mentioned by a few users in the extensions store.

**To Reproduce**
1. start your GNOME session
2. enable isolation in GNOME settings
3. enable isolation in Customized Workspaces
4. open a terminal window on workspace 1
5. switch to workspace 2
6. try to open another terminal from dock or Activities

**Expected behavior**
it should open a new terminal window

**Comment by blipk** [comment]
@khimaros 
Please provide your debug.log file and a list of your other installed extensions.

As I explained to the other user in the extensions website, I'm fairly sure this is a regression in other extensions and/or gnome-shell

**Comment by khimaros** [comment]
i tried with all other extensions disabled. it reproduces, so i suspect it is either a bug in GNOME shell or it is a change in the API for working with them.

**Comment by blipk** [comment]
Works fine for me on Gnome44.
I have no way to go back to 43 to debug this, and don't particularly feel like doing so.
I suggest you upgrade

**Comment by khimaros** [comment]
i've just tested with gnome-shell 44.3 and it is still broken there

**Comment by khimaros** [comment]
```
$ gnome-extensions list --enabled
worksets@blipk.xyz

$ gnome-shell --version
GNOME Shell 44.3

$ dconf dump /org/gnome/shell/extensions/worksets/
[/]
cli-switch=''
debug-mode=false
disable-wallpaper-management=true
hide-app-list=true
isolate-workspaces=true
only-background-details=false
reverse-menu=false
show-helpers=false
show-notifications=false
show-panel-indicator=true
show-workspace-overlay=false
```

**Comment by khimaros** [comment]
with logging enabled, this is the error i see when toggling isolation on:

```
(worksets@blipk.xyz) [8/25/2023, 2:33:02 AM]:!Error   | () => | 
|-TypeError Me.gExtensions.dash2dock is undefined
|-Stack Trace:
activateIsolater@~/.local/share/gnome-shell/extensions/worksets@blipk.xyz/workspaceManager.js:276:28
apply@~/.local/share/gnome-shell/extensions/worksets@blipk.xyz/panelIndicator.js:108:57
_buildOptionsMenuItems/</optionMenuItem.pressHandler<@~/.local/share/gnome-shell/extensions/worksets@blipk.xyz/panelIndicator.js:134:85
toggle@resource:///org/gnome/shell/ui/popupMenu.js:431:14
_buildOptionsMenuItems/</optionMenuItem.activate@~/.local/share/gnome-shell/extensions/worksets@blipk.xyz/panelIndicator.js:136:101
vfunc_button_release_event@resource:///org/gnome/shell/ui/popupMenu.js:146:14
```

does this functionality have a dependency on the dash2dock extension?

**Comment by khimaros** [comment]
@blipk kindly requesting to re-open this issue.

**Comment by khimaros** [comment]
i've confirmed that the problem is contained within this block by commenting out everything except for the builtin isolation support.

https://github.com/khimaros/Customised-Workspaces/blob/master/worksets%40blipk.xyz/workspaceManager.js#L270

somehow this conditional is evaluating to true on a machine without any dash2dock plugin installed nor anything with such a key name in dconf:

```python
else if (Me.gExtensions?.dash2dock.settings && Me.gExtensions?.dash2dock.state === extensionUtils.ExtensionState.ENABLED) {
```

quite odd because trying this expression in Looking Glass, it returns undefined.

this is the workaround: https://github.com/blipk/Customised-Workspaces/compare/master...khimaros:Customised-Workspaces:master

**Comment by blipk** [comment]
Hey @khimaros 
For some reasons I wasn't getting notifications for this, thanks for digging into the issue.

I thought I had fixed this rpeviously, but it seems I misplaced the optional chaining operator on `gExtensions?` when it should have been on `dash2dock?`

There's no dependency on dash2dock or dash2panel, but they provide their own workspace isolation so my extension has to check theirs to make sure they're not conflicting.

I've posted an updated fix to the extensions store that should be available in a few days.
